transformers/storage/median/repository.go

```diff
@@ -0,0 +1,286 @@
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+	_, insertErr := tx.Exec(insertMedianBarQuery, diffID, headerID, addressID, bar)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("bar record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func (repository *MedianStorageRepository) insertPackedValueRecord(diffID, headerID int64, contractAddress string, metadata types.ValueMetadata, packedValues map[int]string) error {
+	for order, value := range packedValues {
+		var insertErr error
+		switch metadata.PackedNames[order] {
+		case Val:
+			insertErr = repository.insertMedianVal(diffID, headerID, contractAddress, value)
+		case Age:
+			insertErr = repository.insertMedianAge(diffID, headerID, contractAddress, value)
+		default:
+			panic(fmt.Sprintf("unrecognized median contract storage name in packed values: %s", metadata.Name))
+		}
+		if insertErr != nil {
+			return insertErr
+		}
+	}
+	return nil
+}
+
+func (repository *MedianStorageRepository) insertOrcl(diffID, headerID int64, metadata types.ValueMetadata, contractAddress, orcl string) error {
+	orclAddress, orclErr := getOrclAddress(metadata.Keys)
+	if orclErr != nil {
+		return orclErr
+	}
+
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+
+	orclAddressID, orclAddressErr := shared.GetOrCreateAddress(orclAddress, repository.db)
+	if orclAddressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("median orcl address", orclAddressErr)
+		}
+		return addressErr
+	}
+
+	_, insertErr := tx.Exec(insertMedianOrclQuery, diffID, headerID, addressID, orclAddressID, orcl)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("bud record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func getOrclAddress(keys map[types.Key]string) (string, error) {
+	user, ok := keys[constants.Address]
+	if !ok {
+		return "", types.ErrMetadataMalformed{MissingData: constants.A}
+	}
+	return user, nil
+}
+
+func (repository *MedianStorageRepository) insertBud(diffID, headerID int64, metadata types.ValueMetadata, contractAddress, bud string) error {
+	budAddress, budErr := getBudAddress(metadata.Keys)
+	if budErr != nil {
+		return budErr
+	}
+
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
```

Comment:
nbd but it could be cool to extract a function that does the `GetOrCreateAddress` the first time it's called and returns the id without a DB query in subsequent calls - just so we don't need to go to the DB for the contract address ID on every insert